async_simple/util/ThreadPool.h

```diff
@@ -93,19 +98,28 @@ inline ThreadPool::ThreadPool(size_t threadNum, bool enableWorkSteal)
 
             // If _enableWorkSteal false or work steal failed, wait for a pop
             // task.
-            if (!workerItem.fn && !_queues[id].pop(workerItem)) {
-                break;
-            }
+            if (!workerItem.fn && !_queues[id].pop(workerItem))
+                continue;
 
-            if (workerItem.fn) {
+            if (workerItem.fn)
                 workerItem.fn();
-            }
         }
     };
 
     _threads.reserve(_threadNum);
     for (auto i = 0; i < _threadNum; ++i) {
         _threads.emplace_back(worker, i);
+
+#ifdef __linux__
+        // Run threads per core.
+        cpu_set_t cpuset;
+        CPU_ZERO(&cpuset);
+        CPU_SET(i, &cpuset);
+        int rc = pthread_setaffinity_np(_threads[i].native_handle(),
+                                        sizeof(cpu_set_t), &cpuset);
+        if (rc != 0)
+            std::cerr << "Error calling pthread_setaffinity_np: " << rc << "\n";
```

Comment:
we need a log component？

Reply:
Yeah, I filed an issue at https://github.com/alibaba/async_simple/issues/105